services/client/services_test.go

```diff
@@ -956,6 +967,49 @@ func TestCacheCollisions_Watchables(t *testing.T) {
 	}
 }
 
+func TestLeaderService(t *testing.T) {
+	mc := gomock.NewController(t)
+	defer mc.Finish()
+
+	ld := newTestLeaderGen(mc)
+
+	opts, closer, _ := testSetup(t)
+	defer closer()
+
+	opts = opts.SetLeaderGen(ld)
+	cl, err := NewServices(opts)
+	require.NoError(t, err)
+
+	newElectionOpts := func() services.ElectionOptions {
+		eo, err := services.NewElectionOptions()
+		require.NoError(t, err)
+		return eo
+	}
+
+	sid1 := services.NewServiceID().SetName("s1")
+	sid2 := services.NewServiceID().SetName("s2")
+	eo1 := newElectionOpts()
+	eo2 := newElectionOpts().SetLeaderTimeout(10 * time.Second)
+	eo3 := newElectionOpts().SetResignTimeout(10 * time.Second)
+
+	for _, sid := range []services.ServiceID{sid1, sid2} {
+		for _, eo := range []services.ElectionOptions{eo1, eo2, eo3} {
+			_, err := cl.LeaderService(sid, eo)
+			assert.NoError(t, err)
+		}
+	}
+
+	assert.Equal(t, 6, len(cl.(*client).ldSvcs),
+		"should cache 6 unique client entries")
+}
+
+func newTestLeaderGen(mc *gomock.Controller) LeaderGen {
```

Comment:
Yeah I just needed something that can return a valid non-nil implementation of `services.LeaderService`